src/sentry/integrations/slack/tasks.py

```diff
@@ -128,14 +128,21 @@ def find_channel_id_for_rule(project, actions, uuid, rule_id=None, **kwargs):
 @instrumented_task(
     name="sentry.integrations.slack.search_channel_id_metric_alerts", queue="integrations"
 )
-def find_channel_id_for_alert_rule(organization_id, uuid, data, alert_rule_id=None):
+def find_channel_id_for_alert_rule(organization_id, uuid, data, alert_rule_id=None, user_id=None):
     redis_rule_status = RedisRuleStatus(uuid)
     try:
         organization = Organization.objects.get(id=organization_id)
     except Organization.DoesNotExist:
         redis_rule_status.set_value("failed")
         return
 
+    user = None
+    if user_id:
+        try:
+            user = User.objects.get(id=user_id)
```

Comment:
@wedamija Nit: might be cleaner if you just do `User.objects.filter(id=user_id).first()`

Reply:
That is a bit cleaner... the sort that `first` puts in place should be fine since we just get a single row. I know matt isn't a fan of using first, so i might just leave it as is.

Reply:
Is it because of performance differences? I'm curious what's the difference exactly.

Reply:
In this case there's no difference and it should be fine. In other cases, `.first()` adds a sort, which can break index usage and cause slow queries.